client/blockchain-service/src/types.rs

```diff
@@ -320,3 +322,68 @@ pub enum NewBlockNotificationKind {
         new_best_block: BestBlockInfo,
     },
 }
+
+/// The information needed to register a Forest Storage snapshot.
+#[derive(Debug, Clone, Encode, Decode)]
+pub struct ForestStorageSnapshotInfo {
+    /// The block number at which the Forest Storage snapshot was taken.
+    ///
+    /// i.e. the block number at which the Forest Storage changed from this snapshot
+    /// version due to adding or removing files.
+    pub block_number: BlockNumber,
+    /// The Forest Storage snapshot hash.
+    ///
+    /// This is to uniquely identify the Forest Storage snapshot, as there could be
+    /// snapshots at the same block number, but in different forks.
+    pub block_hash: H256,
+    /// The Forest Storage root when the snapshot was taken.
+    ///
+    /// This is used to identify the Forest Storage snapshot and retrieve it.
+    pub forest_root: HasherOutT<StorageProofsMerkleTrieLayout>,
+}
+
+impl PartialEq for ForestStorageSnapshotInfo {
+    fn eq(&self, other: &Self) -> bool {
+        self.block_number == other.block_number
+            && self.block_hash == other.block_hash
+            && self.forest_root == other.forest_root
+    }
+}
+
+impl Eq for ForestStorageSnapshotInfo {}
+
+impl PartialOrd for ForestStorageSnapshotInfo {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Is there a reason why we can't use `derive` for these implementations?

Reply:
`PartialOrd` I want it to explicitly have the custom behaviour set in `Ord`. The other two, yeah, definitely.